metrics/csv_metrics.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 The Celo Authors
+// This file is part of the celo library.
+//
+// The celo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The celo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the celo library. If not, see <http://www.gnu.org/licenses/>.
+
+package metrics
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+)
+
+// A CSVRecorder enables easy writing of CSV data a specified writer.
+// The header is written on creation. Writing is thread safe.
+type CSVRecorder struct {
```

Comment:
I would say to use https://golang.org/pkg/encoding/csv/ rather than do all the formatting yourself.  Generally speaking it's better to use the library rather than DIY, since CSV has many edge cases, like if one of the values has a comma in it (though that shouldn't be the case here).  Though it does require the row to consist of strings, whereas here we're getting `interface{}`, but that seems manageable.

Given the limited nature and relatively simplicity of the CSV we're generating, I don't feel super-strongly about this, but I think it's a best-practice that's always a good idea to follow (especially when the language has csv support built-in and you don't need an external dependency).

Reply:
Yea, using the the library definitely makes sense.

What do you think about switching the interface to be []string (or varags string) and then calling out the csv library. I think I still want to wrap the library (specifically to be able to eventually close the file).

Reply:
Ok, I've switched to using the csv library. It makes things a little harder b/c there's no way easy way to convert primitive types (like `int64`) to string values outside the fmt package, but it works fine.

Reply:
There is `strconv.FormatInt()` for int64, so we could use that.  But, yeah, we definitely need to convert to strings one way or another.  I don't have a strong preference on whether it should be inside or outside `CSVRecorder`.